### PR TITLE
Only launch VM specified in test set in the CI

### PR DESCRIPTION
**Pull request by usamasaqib**
### What does this PR do?
This PR makes the configuration generation test set aware. This means we only launch VMs required by a test set as specified in the Gitlab yaml. This allows us to reduce resource usage by unused VMs.

### CODE REVIEW

**Review by paulcacheux — APPROVED** [review]
LGTM for agent-security files

File: tasks/kernel_matrix_testing/platforms.py [hunk 1]
```diff
@@ -26,7 +26,15 @@ def get_platforms():
         return cast("Platforms", json.load(f))
 
 
-def filter_by_ci_component(platforms: Platforms, component: Component) -> Platforms:
+class KMTTestJob:
+    def __init__(self, name, arch, test_set, kernels):
```
Comment (gjulianm) [inline tasks/kernel_matrix_testing/platforms.py:30]: Could we have type annotations for these values? Specially for the arch one, to avoid mistakes later.

**Review by gjulianm — APPROVED** [review]
LGTM, just a minor nit.

**Pull request merged**